Add `swift package add-target-plugin` command

hi2gage opened the pull request:
Adds `swift package add-target-plugin` command for adding plugins to target from CLI

### Motivation:
Featured requested in #8169 

Allows users to add package plugins to targets via command line. 

[Forums Discussion](https://forums.swift.org/t/proposal-swift-package-add-target-plugin-command-to-swiftpm/77930)

### Modifications:
Add new `add-target-plugin` command which aligns with the existing `add-target-dependency` command


```bash
> swift package add-target-plugin --help
OVERVIEW: Add a new target plugin to the manifest

USAGE: swift package add-target-plugin <plugin-name> <target-name> [--package <package>]

ARGUMENTS:
  <plugin-name>           The name of the new plugin
  <target-name>           The name of the target to update

OPTIONS:
  --package <package>     The package in which the plugin resides
  --version               Show the version.
  -h, -help, --help       Show help information.
```


### Result:
After running the following command:

```bash
swift package add-target-plugin OpenAPIGenerator myTarget --package swift-openapi-generator
```

The following addition will be made to the `Package.swift` manifest:

```swift
.target(
  name: "myTarget",
  plugins: [
    .plugin(name: "OpenAPIGenerator", package: "swift-openapi-generator")
  ]
)
```

hi2gage commented on Tests/PackageRegistryTests/RegistryClientTests.swift:
```diff
@@ -803,7 +803,7 @@ final class RegistryClientTests: XCTestCase {
         let availableManifests = try await registryClient.getAvailableManifests(
             package: identity,
             version: version,
-            observabilityScope: ObservabilitySystem.NOOP,
```
Fixed borked tests in main

dschaefer2 commented:
@swift-ci please test windows self hosted

dschaefer2 commented:
@swift-ci please test self hosted windows

bkhouri commented:
@swift-ci please test windows

hi2gage commented:
@dschaefer2 @bkhouri just want to bump this, I rebased the new changes.

hi2gage commented:
@swift-ci please test self hosted windows

bkhouri commented:
@swift-ci test self hosted windows

bkhouri reviewed (changes requested):
Could I ask that we add a bit more automated tests?

bkhouri commented:
@swift-ci test self hosted windows

bkhouri commented:
Though I don't see how, for some reason, the Windows Self hosted pipeline seems to hang with this change.

bkhouri commented:
@swift-ci test

bkhouri commented:
@swift-ci test self hosted windows

bkhouri commented:
@swift-ci test windows

bkhouri commented:
@swift-ci test

bkhouri commented:
@swift-ci test self hosted windows

bkhouri commented:
@swift-ci test windows

bkhouri commented:
@swift-ci test

bkhouri commented:
@swift-ci test self hosted windows

bkhouri commented:
@swift-ci test

hi2gage commented:
> @hi2gage This looks like it might benefit from the same code as in #8534 to allow `add-target-plugin` to be idempotent when adding a target plugin that already exists.

Hi @plemarquand, I was taking a break from my side projects and finally getting back to this. I was able to make the changes to make `add-target-plugin` to be idempotent. Please let me know what you think.

Thanks!

plemarquand reviewed (approved):
Great, looks good, thank you!!

Commit:
Add swift package add-target-plugin command

Adds a new `swift package add-target-plugin` subcommand that allows
users to add plugin usages to targets in their Package.swift manifest.

Usage:
  swift package add-target-plugin <plugin-name> <target-name> [--package <package>]

Example:
  swift package add-target-plugin OpenAPIGenerator myTarget --package swift-openapi-generator

This command uses SwiftRefactor.AddPluginUsage from swift-syntax to
perform the manifest edits, following the same pattern as other
add-* commands.

Changes:
- Add AddTargetPlugin command that uses SwiftRefactor.AddPluginUsage
- Register add-target-plugin as a subcommand
- Add comprehensive tests covering both internal and external plugins
- Add .Feature.Command.Package.AddTargetPlugin test tag

owenv commented:
This was approved in https://forums.swift.org/t/accepted-se-0511-add-target-plugin-command-to-swift-package-manager/86206. Rerunning CI before I merge since it's been awhile...
